Instruments/bridge12.py

```diff
@@ -250,50 +219,37 @@ def set_power(self,dBm):
             "power to change: I'm trying to set to %d/10 dBm, but the Bridge12"
             "keeps replying saying that it's set to %d/10"
             "dBm")%(setting,result))
-    def rxpowermv_int_singletry(self):
-        """read the integer value for the Rx power (which is 10* the value in mV).  Also has a software interlock so that if the Rx power ever exceeds self.safe_rx_level_int, then the amp shuts down."""
-        self.write(b'rxpowermv?\r')
-        retval = self.readline()
-        retval = retval.strip()
-        if retval == 'ERROR':
-            logger.info("Found ERROR condition on trying to read rxpowermv -- trying again")
-            j = 0
-            while j < 10:
-                time.sleep(10e-3)
-                self.write(b'rxpowermv?\r')
-                retval = self.readline()
-                retval = retval.strip()
-                if retval != 'ERROR':
-                    logger.info("after try %d, it responded with %s"%(j+1,retval))
-                    break
-        retval = int(retval)
-        if retval > self.safe_rx_level_int:
-            raise RuntimeError("Read an unsafe Rx level of %0.1fmV"%(retval/10.))
-        return retval
-    def rxpowermv_float(self):
-        "need two consecutive responses that match"
-        for j in range(3):
-            # burn a few measurements
-            self.rxpowermv_int_singletry()
-        h = self.rxpowermv_int_singletry()
-        i = self.rxpowermv_int_singletry()
-        while h != i:
-            h = self.rxpowermv_int_singletry()
-            i = self.rxpowermv_int_singletry()
-        return float(h)/10.
-    def txpowermv_int_singletry(self):
-        self.write(b'txpowermv?\r')
-        return int(self.readline())
-    def txpowermv_float(self):
+    def rxpowerdbm_float(self):
+        """read the integer value for the Rx power -- loops three times to
+        check a consistent Rx is being read.
+
+        If all three tries are above safe_rx_level_int, triggers a safety interlock."""
+        self.reset_input_buffer()
+        def grab_consist_value():
+            rx_try1 = self.robust_int_response(b'rxpowerdbm?\r')
+            rx_try2 = self.robust_int_response(b'rxpowerdbm?\r')
+            for j in range(20):
+                rx_try3 = self.robust_int_response(b'rxpowerdbm?\r')
+                if abs(rx_try1 - rx_try2)<2 and abs(rx_try2 - rx_try3)<2 and abs(rx_try3-rx_try1)<2:
+                    return (rx_try1+rx_try2+rx_try3)/3.0
+                else:
+                    rx_try1,rx_try2 = rx_try2,rx_try3
+            raise ValueError("I tried 20 times to grab a consistent power, and could not (most recent %f, %f, %f)"%(rx_try1,rx_try2,rx_try3))
```

Comment:
we discussed a PR being made to make a function for the looping to get a consistent value (https://www.wrike.com/open.htm?id=1285925476) . For future reference this is the function we would want just it's own method right?